src/set_theory/game/pgame.lean

```diff
@@ -148,6 +139,45 @@ def move_right : Π (g : pgame), right_moves g → pgame
 @[simp] lemma right_moves_mk {xl xr xL xR} : (⟨xl, xr, xL, xR⟩ : pgame).right_moves = xr := rfl
 @[simp] lemma move_right_mk {xl xr xL xR} : (⟨xl, xr, xL, xR⟩ : pgame).move_right = xR := rfl
 
+/--
+Construct a pre-game from list of pre-games describing the available moves for Left and Right.
+-/
+-- TODO define this at the level of games, as well, and perhaps also for finsets of games.
+def of_lists (L R : list pgame.{u}) : pgame.{u} :=
+mk (ulift (fin L.length)) (ulift (fin R.length))
+  (λ i, L.nth_le i.down i.down.is_lt) (λ j, R.nth_le j.down.val j.down.is_lt)
+
+@[simp] lemma left_moves_of_lists (L R : list pgame) :
+  (of_lists L R).left_moves = ulift (fin L.length) := rfl
+@[simp] lemma right_moves_of_lists (L R : list pgame) :
+  (of_lists L R).right_moves = ulift (fin R.length) := rfl
```

Comment:
Wouldn't we be better off without these? No one wants to have to interact with those `ulift`s. It looks like you're making sure people can avoid them anyway in the next declarations.

Reply:
I think we should have the lemmas, but maybe we'd be better off without them being `simp`

Reply:
Again, I don't see the problem with having these lemmas and making them `simp`. If you're proving something about the types themselves, then chances are these lemmas will help some.

Reply:
I can see one circumstance in which this might potentially be annoying: if a theorem can be proved by providing an equivalence, such as `to_of_lists_left_moves`. But even in that case, providing the equivalence between `ulift (fin _)` and `fin _` is trivial. So yeah, I'd rather keep the `simp` just in case.

Reply:
Sorry, I disagree, and still think you should remove the `@[simp]`s here.

(And providing the equivalence between ulift fin and fin may be trivial, but it would also be annoying if unnecessary!)

Reply:
Well, I'm not currently using them anyway, so I guess I'll remove it. But if I ever need them, I'll bring this up again!

Reply:
Absolutely! :-)